prow/plugins/trigger/generic-comment.go

```diff
@@ -197,20 +198,36 @@ func FilterPresubmits(honorOkToTest bool, gitHubClient GitHubClient, body string
 	return pjutil.FilterPresubmits(filter, changes, branch, presubmits, logger)
 }
 
-func availablePresubmits(githubClient GitHubClient, body, org, repo, branch string, number int, presubmits []config.Presubmit, logger *logrus.Entry) ([]string, error) {
+// availablePresubmits returns 2 sets of presubmits:
+// 1. presubmits that can be run with '/test all' command.
+// 2. presubmits that can be run with their trigger, e.g. '/test job'
+func availablePresubmits(githubClient GitHubClient, body, org, repo, branch string, number int, presubmits []config.Presubmit, logger *logrus.Entry) ([]string, []string, error) {
 	changes := config.NewGitHubDeferredChangedFilesProvider(githubClient, org, repo, number)
-	all := func(p config.Presubmit) (bool, bool, bool) {
-		return true, true, true
+
+	runWithTestAll, err := pjutil.FilterPresubmits(pjutil.TestAllFilter(), changes, branch, presubmits, logger)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var triggerFilters []pjutil.Filter
+	for _, ps := range presubmits {
```

Comment:
This needs to filter out presubmits that can not run via `ps.CouldRun(branch)`

Reply:
Thank you for the review. I reviewed the code again and it seems `FilterPresubmits` already excludes those by calling `ShouldRun` on each presubmit job. I have updated the PR and added a test case to make sure this is covered.